tools/testsys/src/main.rs

```diff
@@ -34,16 +34,23 @@ struct TestsysArgs {
 
     /// Path to the kubeconfig file for the testsys cluster. Can also be passed with the KUBECONFIG
     /// environment variable.
-    #[clap(long)]
+    #[clap(long, env = "TESTSYS_KUBECONFIG_PATH")]
     kubeconfig: Option<PathBuf>,
 
+    /// If the provided kubeconfig file doesn't exist, infer the kubeconfig from the environment.
+    #[clap(long, short)]
+    default: bool,
+
     #[clap(subcommand)]
     command: Command,
 }
 
 impl TestsysArgs {
     async fn run(self) -> Result<()> {
-        let client = match self.kubeconfig {
+        let kubeconfig_path = self
+            .kubeconfig
+            .filter(|kubeconfig| !self.default || kubeconfig.exists());
```

Comment:
Following our offline discussion:

I didn't read the code properly. Now it makes senses but I have a different complain, even though `-d` is always set, it will be ignored if `TESTSYS_KUBECONFIG_PATH` exists, that makes me feel like the CLI isn't respecting the instructions set with the `testsys -d` command, which clearly says use the default.

If your intention is to always use `TESTSYS_KUBECONFIG_PATH` regardless of the flags, then I think what I suggested fits better: don't add a flag, and just express the precedence.

// pseudo-code
if path.is_some() && path.exists() {
    TestManager::from_file(&path)
} else {
     TestManager::new();
}

With something similar to ^, you respect the instruction given in the command:

```bash
# Use env variable
export TESTSYS_KUBECONFIG_PATH=here; testsys logs --test

# Use default
testsys logs --test
```

It is easier to document as well, otherwise the documentation will be like "if ENV is set, it will be used even if `-d` is passed" (I don't like when CLIs aren't clear with their flags :sweat_smile:)

Reply:
The only problem with that is it may not do what the user expects.
Assume my kubeconfig is `my-kubeconfig.yaml`.
If I ran `testsys --kubeconfig my_kubeconfig.yaml status` (instead of  `testsys --kubeconfig my-kubeconfig.yaml status`), it would use my default context which I clearly didn't want since I specified a kubeconfig.

Reply:
Not quite, `TESTSYS_KUBECONFIG_PATH` is mapped to the `kubeconfig` option, so `self.kubeconfig` will be "something" and technically these two will be interchangeable:

```bash
# This won't use the default
testsys --kubeconfig <path-to-config> <blah>
# Neither will this
export TESTSYS_KUBECONFIG_PATH=<path-to-config> testsys <blah>
```

And only this will use `default`, without any flags passed:
```bash
# `kubeconfig` isn't set, so use the default kubeconfig
testsys <blah>
```

And ^ still mimics what other CLIs do: use `$HOME/.kube/config` unless `--kubeconfig` (or in your case `TESTSYS_KUBECONFIG_PATH`) is set.

Reply:
How would that work with `cargo make` having a default kubeconfig location set? Then cargo make would have to check to see if `TESTSYS_KUBECONFIG_PATH` exists. If it exists pass it into the command if not ignore it. Is that the implementation you would prefer? I prefer the command handling the file path logic in the CLI instead of Makefile scripts.

Reply:
I agree with @arnaldo2792 - I wouldn't expect to have to pass a flag to use the default. I also would say it's more robust to have `testsys` get a known-good set of arguments, and to error out if told to use a kubeconfig that doesn't exist, since otherwise that could mask a misconfiguration.

Since `cargo make` is invoking `testsys` and sort of owns deciding how to invoke based on the repo state, it makes sense for it to test whether `${BUILDSYS_ROOT_DIR}/testsys.kubeconfig` exists, and if it doesn't, just not pass that argument.

I might like to generalize it a bit and populate `CARGO_MAKE_TESTSYS_ARGS`, and then just always pass that field into `testsys` invocations - similar to how `CARGO_MAKE_CARGO_ARGS` works.

(You can also add `TESTSYS_ARGS` at the same time so there's a variable that's ready to go for overrides that won't change the default args - similar to `CARGO_BUILD_ARGS`.)